app/gui/src/model/execution_context/synchronized.rs

```diff
@@ -512,4 +532,59 @@ pub mod test {
             context.modify_visualization(vis_id, expression, module).await.unwrap();
         });
     }
+
+    /// Check that the [`ExecutionContext::load_component_groups`] method correctly parses a mocked
+    /// Language Server response and loads the result into a field of the [`ExecutionContext`].
+    #[test]
+    fn loading_component_groups() {
+        fn library_component(name: &str) -> language_server::LibraryComponent {
+            language_server::LibraryComponent { name: name.to_string(), shortcut: None }
+        }
+        let sample_ls_component_groups = vec![
+            language_server::LibraryComponentGroup {
+                library: "local.Unnamed_10".to_string(),
+                group:   "Test Group 1".to_string(),
+                color:   Some("#C047AB".to_string()),
+                icon:    None,
+                exports: vec![
+                    library_component("Standard.Base.System.File.new"),
+                    library_component("local.Unnamed_10.Main.main"),
+                ],
+            },
```

Comment:
This test is really long and complex. Please either split it down to several tests or add docs inside to document what it does.